DevEdu.API/DevEdu.API/Controllers/CourseController.cs

```diff
@@ -28,40 +34,50 @@ public class CourseController : Controller
             _courseRepository = courseRepository;
             _courseService = courseService;
             _mapper = mapper;
+            //_claimsIdentity = this.User.Identity as ClaimsIdentity;
         }
 
         [HttpGet("{id}/simple")]
-        [Description("Get course by id with groups")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status200OK)]
-        public CourseInfoFullOutputModel GetCourseSimple(int id)
+        [Description("Get course by id with topics")]
+        [Authorize]
+        [ProducesResponseType(typeof(CourseInfoShortOutputModel), StatusCodes.Status200OK)]
+
+        public CourseInfoShortOutputModel GetCourseSimple(int id)
         {
             var course = _courseService.GetCourse(id);
-            return _mapper.Map<CourseInfoFullOutputModel>(course);
+            return _mapper.Map<CourseInfoShortOutputModel>(course);
         }
 
         [HttpGet("{id}/full")]
         [Description("Get course by id full")]
+        [AuthorizeRoles(Role.Admin, Role.Manager, Role.Teacher, Role.Methodist)]
         [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status200OK)]
+
         public CourseInfoFullOutputModel GetCourseFull(int id)
         {
-            var course = _courseService.GetFullCourseInfo(id);
+            var userToken = this.GetUserIdAndRoles();
+            var course = _courseService.GetFullCourseInfo(id, userToken);
             return _mapper.Map<CourseInfoFullOutputModel>(course);
         }
 
         [HttpGet]
         [Description("Get all courses")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status200OK)]
-        public List<CourseInfoFullOutputModel> GetAllCoursesWithGrops()
+        [Authorize]
+        [ProducesResponseType(typeof(CourseInfoShortOutputModel), StatusCodes.Status200OK)]
+
+        public List<CourseInfoShortOutputModel> GetAllCoursesWithGrops()
         {
             var courses = _courseService.GetCourses();
-            return _mapper.Map<List<CourseInfoFullOutputModel>>(courses);
+            return _mapper.Map<List<CourseInfoShortOutputModel>>(courses);
         }
 
 
         [HttpPost]
         [Description("Create new course")]
-        [ProducesResponseType(typeof(CourseInfoFullOutputModel), StatusCodes.Status201Created)]
-        public CourseInfoFullOutputModel AddCourse([FromBody] CourseInputModel model)
+        [AuthorizeRoles(Role.Admin, Role.Manager, Role.Teacher, Role.Methodist)]
```

Comment:
and here as well
in addition, Teacher can't create a course

Reply:
+